includes/Embed_Block.php

```diff
@@ -211,4 +229,81 @@ protected function render_block_feed( array $attributes ) {
 
 		return (string) ob_get_clean();
 	}
+
+	/**
+	 * Filters whether AMP should be skipped for the current post.
+	 *
+	 * Disables AMP for the <amp-iframe> proxy requests.
+	 *
+	 * @param bool $skip Whether to skip AMP for the current post.
+	 *
+	 * @return bool Whether to skip AMP or not.
+	 */
+	public function skip_amp_for_proxy_request( $skip ) {
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		if ( isset( $_GET['_web_story_embed_proxy'] ) && has_block( self::BLOCK_NAME ) ) {
+			return true;
+		}
+
+		return $skip;
+	}
+
+	/**
+	 * Renders the block type output in AMP context.
+	 *
+	 * @param array $attributes Block attributes.
+	 * @param int   $count Counter used for the proxy URL.
+	 *
+	 * @return string Rendered block type output.
+	 */
+	protected function render_block_amp( array $attributes, $count ) {
+		$url   = (string) $attributes['url'];
+		$title = (string) $attributes['title'];
+		$align = sprintf( 'align%s', $attributes['align'] );
+
+		$iframe_url = add_query_arg(
+			[
+				'_web_story_embed_proxy' => $count,
+			]
+		);
+
+		ob_start();
+		?>
+		<div class="wp-block-web-stories-embed <?php echo esc_attr( $align ); ?>">
+			<amp-iframe
+				width="<?php echo esc_attr( $attributes['width'] ); ?>"
+				height="<?php echo esc_attr( $attributes['height'] ); ?>"
+				sandbox="allow-scripts"
```

Comment:
In order to follow links, this may need:

```suggestion
				sandbox="allow-scripts allow-popups allow-popups-to-escape-sandbox allow-top-navigation"
```